### PR TITLE
Message makes tree diameter too narrow to use

### PR DESCRIPTION
**Pull request by RobinIsTheBird**
@lederer recommended we display the message beneath the field label because there is plenty of room there.

Also, the message should be in red, like the message in the add-a-tree sidebar wizard.

Also, the label should not be vertical-align middle. Its baseline should be even with the baseline of the text in the second table cell, whether it's in edit mode or not.

This needed to be done for all rows in the detail pages. Otherwise, the presence of a message on the right on any row would make the tree diameter too narrow to use.

Changes:

opentreemap/treemap/js/src/lib/
------------------------------------------
* editableForm.js
* inlineEditForm.js
* simpleEditForm.js

Manipulate a [data-mode] element to be either edit or display.

opentreemap/treemap/templates/treemap/partials/
------------------------------------------------------------------
* map_feature_detail_base.html

Set #map-feature-form to have a data-mode attribute

* plot_detail.html
* resource_detail.html

Add a can-inline-edit class to the tables to distinguish them from ordinary tables for styling purposes. Specifically, it distinguishes them from the diameter-circumference control, which uses table layout.

opentreemap/treemap/templates/treemap/field/
--------------------------------------------------------------
* attrs.html
* tr.html
* diameter_tr.html
* species_tr.html

Add a css_class variable to attrs.html to use in the class attribute, as distinct from the class variable already in use for the data-class attribute.

The *tr.html templates set css_class to `alert alert-danger` on the error elements to make them styled with red.

assets/css/sass/partials/pages/_treedetails.scss
------------------------------------------------

Remove the diameter-row style.
Style the first td in a row differently depending on whether
* the table is in edit mode
* the table contents can be edited inline
* the row the td is in is itself editable

Attention to whether the table contents can be edited allows the style to apply just to the desired table's cells, not to cells in child tables such as the diameter-circumference editor.

--

Connects to #2754

### CODE REVIEW

File: opentreemap/exporter/tests.py [hunk 1]
```diff
@@ -287,39 +287,39 @@ def test_export_users_json_keep_info_private(self):
...
-                             'last_edit_model_id': str(self.plot.pk),
-                             'last_edit_instance_id': str(self.instance.pk),
-                             'last_edit_user_id': str(self.user1.pk),
-                             'email': 'genly@example.com',
-                             'email_hash': self.user2.email_hash,
-                             'last_edit_model': 'Tree',
-                             'last_edit_model_id': str(self.tree.pk),
-                             'last_edit_instance_id': str(self.instance.pk),
-                             'last_edit_user_id': str(self.user2.pk)})
+#    def test_export_users_json_make_info_public(self):
```
Comment (jwalgran) [inline opentreemap/exporter/tests.py]: The assert in this test does work, but has what looks like some copy/paste or merge cruft.

Python will let you redefine keys when declaring a literal map. The last value wins.

```python
In [1]: d = {'a':1 , 'b': 2, 'a': 3}

In [2]: d
Out[2]: {'a': 3, 'b': 2}
```

The literal hash can be fixed to remove redefined keys, and the test still passes.

```diff
diff --git a/opentreemap/exporter/tests.py b/opentreemap/exporter/tests.py
index c08c3c8..ab9235b 100644
--- a/opentreemap/exporter/tests.py
+++ b/opentreemap/exporter/tests.py
@@ -310,16 +310,12 @@ class UserExportsTest(UserExportsTestCase):
                              'created': str(self.user1.created)})
 
         self.assertUserJSON(user2data,
-                            {'last_edit_model': 'Plot',
-                             'last_edit_model_id': str(self.plot.pk),
-                             'last_edit_instance_id': str(self.instance.pk),
-                             'last_edit_user_id': str(self.user1.pk),
-                             'email': 'genly@example.com',
-                             'email_hash': self.user2.email_hash,
-                             'last_edit_model': 'Tree',
+                            {'last_edit_model': 'Tree',
                              'last_edit_model_id': str(self.tree.pk),
                              'last_edit_instance_id': str(self.instance.pk),
-                             'last_edit_user_id': str(self.user2.pk)})
+                             'last_edit_user_id': str(self.user2.pk),
+                             'email': 'genly@example.com',
+                             'email_hash': self.user2.email_hash})
 
     def test_min_edit_date(self):
         last_week = now() - datetime.timedelta(days=7)
```

I tried both commenting out and fixing this test and it did not resolve the hard to pin down memory pressure issues we are currently running into.

I would prefer that we not include this commit in the PR.

**Comment by jwalgran** [comment]
The Jenkins build failure is caused by the fact that we can no longer mix and match the `develop` branches of repos with the `release/2.11.0` branches. The reviewer can run the tests with all of the `release/2.11.0` branches checked out to verify that there are no regressions.

File: opentreemap/exporter/tests.py (hunk 1 above)
Comment (RobinIsTheBird) [inline opentreemap/exporter/tests.py]: Not saying the test had been failing, but that it was passing in spite of errors in the test.

If the repetition was caused by merge kruft, it was caused two years ago. Simply an incorrect test, whether it fails or passes.

Meanwhile, I can't get through travis without this change.

**Comment by RobinIsTheBird** [comment]
Submitting this PR for code review in spite of ci failures, because the failures are unrelated to my changes.

Commenting a test out of exporter.tests gets past travis, but that doesn't belong in this PR, so I took it out.

Jenkins fails trying to provision because of a statsd incompatibility between develop and release. I don't understand why release is pulling in that change, but at this point it's all beyond me.

Apart from transient memory issues, also apparently unrelated to my changes, tests pass on my local vm.

**Pull request closed without merging**